### DISCUSSION TITLE
[FEATURE REQUEST] extend the sources beyond API for queries on SQLite / MySQL

### DISCUSSION THREAD

**Discussion by tlwt** [Rooyca/obsidian-api-request#13] [Ideas]
I think the API Plugin provides great functionality which could be extended to other sources like SQLite or MySQL
The difference is that it may return multiple rows so that a dedicated header format. Alternatively those could be dropped completey and derived from the query.

```
url: sqlite3:path/file.db
query: SELECT name, description, group FOM demo_table WHERE group = "demo"
format-header: | name  | description | group |
format: | {} | {} | {} |
```
an alternative URL could be `mysql://username:password@someHost.com:3306/database`

**Comment by Rooyca** [comment]
I like the idea but I thinks this would make more sense as a separate plugin. 
Looks like nobody is working on something like this, so I'll start when I have some free time.

**Reply by Rooyca** [reply]
[db2rest](https://github.com/kdhrubo/db2rest) or [resquel](https://github.com/formio/resquel) will convert your DB into a REST API (Don't work with SQLITE).

For SQLITE you could code a simple python app.

**UNSAFE CODE ALERT**

```python
from flask import Flask, jsonify, request
import sqlite3, os

app = Flask(__name__)

# Function to connect to the SQLite database
def connect_db(db_path):
    return sqlite3.connect(db_path)

# Endpoint to execute SQL queries
@app.route('/api/query', methods=['POST'])
def execute_query():
    try:
        query = request.json['query']
        db_path = request.json['db']
    except KeyError:
        return jsonify({'error': 'Parameter missing'}), 400

    try:
        db_path = os.path.join(os.path.dirname(__file__), db_path)
        conn = connect_db(db_path)
        cursor = conn.cursor()
        cursor.execute(query)
        result = cursor.fetchall()
        conn.commit()
        cursor.close()
        conn.close()
        return jsonify({'result': result})
    except sqlite3.Error as e:
        return jsonify({'error': str(e)}), 400

if __name__ == '__main__':
    app.run(debug=True)
```
Maybe that can helps for now.
